js/packages/openinference-core/test/trace_config/OITracer.test.ts

```diff
@@ -0,0 +1,131 @@
+import { OITracer, REDACTED_VALUE } from "@core/trace";
+import { OISpan } from "@core/trace/trace_config/OISpan";
+import {
+  context,
+  Span,
+  SpanKind,
+  SpanOptions,
+  Tracer,
+} from "@opentelemetry/api";
+
+describe("OITracer", () => {
+  let mockTracer: jest.Mocked<Tracer>;
+  let mockSpan: jest.Mocked<Span>;
+
+  beforeEach(() => {
+    mockSpan = {
+      setAttribute: jest.fn(),
+      setAttributes: jest.fn(),
+    } as unknown as jest.Mocked<Span>;
+
+    mockTracer = {
+      startSpan: jest.fn(() => mockSpan),
+      startActiveSpan: jest.fn((name, options, context, fn) => {
+        return fn(mockSpan);
+      }),
+    } as unknown as jest.Mocked<Tracer>;
+  });
+
+  describe("startSpan", () => {
+    it("should create an OISpan with start span and set attributes according to the trace config", () => {
+      const oiTracer = new OITracer({
+        tracer: mockTracer,
+        traceConfig: {
+          hideInputs: true,
+        },
+      });
+      const name = "test-span";
+      const options = {
+        attributes: { key1: "value1", "input.value": "sensitiveValue" },
+      };
+
+      const span = oiTracer.startSpan(name, options);
+
+      expect(mockTracer.startSpan).toHaveBeenCalledWith(
+        name,
+        { attributes: undefined },
+        undefined,
+      );
+      expect(mockSpan.setAttributes).toHaveBeenCalledWith({
+        key1: "value1",
+        "input.value": REDACTED_VALUE,
+      });
+
+      expect(span).toBeInstanceOf(OISpan);
+    });
+  });
+
+  describe("startActiveSpan", () => {
+    it("should create an OISpan with startActiveSpan and set attributes according to the trace config", () => {
+      const oiTracer = new OITracer({
+        tracer: mockTracer,
+        traceConfig: {
+          hideInputs: true,
+        },
+      });
+      const name = "test-span";
+      const options = {
+        attributes: { key1: "value1", "input.value": "sensitiveValue" },
+      };
+
+      const span = oiTracer.startActiveSpan(name, options, (span) => {
+        return span;
+      });
+
+      expect(mockTracer.startActiveSpan).toHaveBeenCalledWith(
+        name,
+        {
+          attributes: undefined,
+        },
+        expect.any(Object),
+        expect.any(Function),
+      );
+      expect(mockSpan.setAttributes).toHaveBeenCalledWith({
+        key1: "value1",
+        "input.value": REDACTED_VALUE,
+      });
+
+      expect(span).toBeInstanceOf(OISpan);
+    });
+  });
+  it("should handle overloads correctly", () => {
```

Comment:
did you mean overrides?

Reply:
no i meant the overloads for this function 
(name, fn)
(name, opts, fn)
(name, opts, context, fn)